refactor(estree): remove regular expression types from ESTree AST

overlookmotel opened the pull request:
`Pattern` and other types in `oxc_regular_expression` are not part of ESTree AST. Don't implement `ESTree`, and don't generate raw transfer deserializers for these types. None of that code is actually used.

#10834 moved `Pattern` to an optional field of `RegExpPattern`, which is skipped in ESTree serialization, which enables now removing it from ESTree AST entirely.

This reduces the scope of ESTree serialization.

Boshen reviewed (approved):
I'll let @leaysgur take a final look.

leaysgur reviewed (approved):
I don't think there's any particular issue as a decision.
(BTW, I'm probably the only person in the world who uses [this](https://github.com/leaysgur/js-regexp-ast-viewer/blob/main/src/oxc_regular_expression_wasm/src/lib.rs#L19), so I might need to think about how to handle. 😅)

I'll also update #10846 accordingly.

Boshen commented:
### Merge activity

* **May 7, 7:50 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(estree): remove regular expression types from ESTree AST (#10855)

`Pattern` and other types in `oxc_regular_expression` are not part of ESTree AST. Don't implement `ESTree`, and don't generate raw transfer deserializers for these types. None of that code is actually used.

This reduces the scope of ESTree serialization.

Boshen commented:
oops the other branch got merged first.

ArnaudBarre commented:
Thanks for cleaning generated types 🎉
